### ISSUE TITLE
Exception is thrown when I want to create a link from the first word of the first paragraph

### ISSUE THREAD

**Issue by orbanbotond** [yabwe/medium-editor#935]
Whenever I select the first word of the first paragraph and I want to turn it to a link nothing happens and instead I receive a JS exception.

This is how the editable is instantiated:
```
    $elem.editMeeteorWay({toolbar: {buttons: ['bold', 'italic', 'underline', 'anchor']},
                          autoLink: true,
                          anchorPreview: false,
                          extensions: {
                              'new-anchor-preview': new NewAnchorPreview(),
                              'edit-anchor-on-click' : new EditAnchorOnClick()
                          }
                        });
```

This is the markup:
```
<div class="editable" data-editor-options="{&quot;placeholder&quot;:{&quot;text&quot;:&quot;Add an agenda item by typing here.&quot;}}" data-insert-path="[&quot;agenda_item&quot;,&quot;agenda_item_text&quot;]" data-params="{&quot;agenda_item&quot;:{&quot;user_id&quot;:2,&quot;meeting_id&quot;:995},&quot;project_id&quot;:64,&quot;object_id&quot;:70119390241700}" data-url="/agenda_items/2405" contenteditable="true" spellcheck="true" data-medium-editor-element="true" role="textbox" aria-multiline="true" medium-editor-index="0" data-placeholder="Add an agenda item by typing here.">
      <p>one two<br><a href="transindex.hu">three</a><br></p><p><a href="amazon.com">one</a> two three</p>
</div>
```
This is the js exception trace:
Uncaught TypeError: Cannot read property 'toLowerCase' of undefinedUtil.getFirstSelectableLeafNode @ medium-editor.js?body=1:1258Selection.importSelectionMoveCursorPastBlocks @ medium-editor.js?body=1:1821Selection.importSelection @ medium-editor.js?body=1:1732MediumEditor.importSelection @ medium-editor.js?body=1:6474MediumEditor.restoreSelection @ medium-editor.js?body=1:6478MediumEditor.extensions.form.extend.completeFormSave @ medium-editor.js?body=1:3359MediumEditor.extensions.form.extend.doFormSave @ medium-editor.js?body=1:3355MediumEditor.extensions.form.extend.handleTextboxKeyup @ medium-editor.js?body=1:3423

**Comment by nmielnik** [comment]
I was able to reproduce this locally just using the innerHTML provided in the example. Looking into it now...

**Comment by orbanbotond** [comment]
Thanks...
I solved temporarily by commenting out:

            // if (typeof selectionState.emptyBlocksIndex !== 'undefined') {
            //     range = this.importSelectionMoveCursorPastBlocks(doc,
root, selectionState.emptyBlocksIndex, range);
            // }

**Comment by nmielnik** [comment]
@orbanbotond I found the root cause and have a fix read in #937

I also found that the reason medium-editor was running into an error was related to the lingering white space elements that your example had in it within the editor but before the first `<p>`.  In your example, you have a newline between the closing tag for the editor, and the first `<p>`, which gets converted into empty text nodes in some browsers,

If you can remove the newlines from the html in the example you provide, I believe you'll be able to work around the medium-editor issue temporarily without having to modify medium-editor code:

Instead of: 
```html<div class="editable" ...>
      <p>one two<br><a href="transindex.hu">three</a><br></p><p><a href="amazon.com">one</a> two three</p>
```

Try this:
```html
<div class="editable" ...><p>one two<br><a href="transindex.hu">three</a><br></p><p><a href="amazon.com">one</a> two three</p></div>
```

Once we cut a release with the new fix, I'll close this issue so you should be able to know when you can upgrade and get the fix that way.

**Comment by nmielnik** [comment]
@orbanbotond thanks again for pointing this issue out, while trying to fix it I found a bunch of other these that needed to be corrected.

Once people have a chance to look over #937 the fix will be in and I'll get a release cut.  I'll ping you once it's there so that you can (hopefully) verify that it fixed your issue.  Thanks!

**Comment by orbanbotond** [comment]
When will this be available publicly and on which branch?

**Comment by j0k3r** [comment]
@orbanbotond this is available in the master branch and will be available in the next 5.14.1 release (in few days I guess).

**Comment by nmielnik** [comment]
@orbanbotond 5.14.1 is out and contains the fix for this.

**Comment by nmielnik** [comment]
@orbanbotond if you're able to find out if your issues is resolved in 5.14.1, will you let us know?

**Comment by orbanbotond** [comment]
Hi Nate,

Thank you for your interest :)
Thank you very much for the fix. It solved our issues ;)
